### PR TITLE
Inspector support

### LINKED ISSUE #1120: support chome devtools
[V8 has extensive debugging abilities built-in](https://github.com/denoland/deno_third_party/blob/56c4acce2e8ffe979b2e7d52d2b3e6f613ed492e/v8/include/v8-inspector.h#L239-L293).
This is what allows Node to connect to Chrome inspector for debugging.
 
The behavior of `node --inspect --debug-brk script.ts` is usually the behavior people want - which is start V8Inspector and break on line 1 of the main script. In deno, I'd like to shorten this to `deno script.ts --debug`.

This is a large feature which will probably require several passes. The core binding to V8Inspector should be done in libdeno.

### PR DESCRIPTION
**Pull request by ry**
Fixes #1120

### CODE REVIEW

File: cli/inspector.rs [hunk 1]
```diff
@@ -0,0 +1,549 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+#![allow(dead_code)]
```
Comment (bartlomieju) [inline cli/inspector.rs]: Remove top-level attribute before landing and use inline ones if needed

File: cli/inspector.rs [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+      mpsc::unbounded_channel::<FrontendToInspectorMsg>();
+    let uuid = Uuid::new_v4();
+
+    let inspector = crate::inspector::DenoInspector::new(
+      scope,
+      context,
+      frontend_to_inspector_rx,
+    );
+
+    eprintln!(
```
Comment (bartlomieju) [inline cli/inspector.rs]: Probably want to respect `--quiet` flag so `info!()` should be used here

File: cli/inspector.rs [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+
+    inspector
+  }
+}
+
+impl Drop for InspectorServer {
+  fn drop(&mut self) {
+    self.server_msg_tx.take();
+    self.thread_handle.take().unwrap().join().unwrap();
+    panic!("TODO: this drop is never called");
```
Comment (bartlomieju) [inline cli/inspector.rs]: Should this be fixed before landing?

File: cli/inspector.rs [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+            "webSocketDebuggerUrl": url,
+          })
+        }).collect();
+        warp::reply::json(&json!(json_values))
+      });
+
+  let version = warp::path!("json" / "version").map(|| {
+    warp::reply::json(&json!({
+      "Browser": format!("Deno/{}", crate::version::DENO),
+      // TODO upgrade to 1.3? https://chromedevtools.github.io/devtools-protocol/
```
Comment (bartlomieju) [inline cli/inspector.rs]: This comment is no longer relevant

File: cli/inspector.rs (hunk 3 above)
Comment (ry) [reply cli/inspector.rs]: This is a rusty_v8 bug. It’s a little strange because a panic would indicate proper operation. I think we should keep it until this is fixed.
